controllers/containerimagebuild_controller.go

```diff
@@ -153,91 +49,49 @@ func (r *ContainerImageBuildReconciler) SetupWithManager(mgr ctrl.Manager) error
 		Complete(r)
 }
 
-// TODO: move the following 2 funcs off of the Reconciler and give them their own k8s client
-
-func (r *ContainerImageBuildReconciler) buildRegistryConfig(ctx context.Context, apiRegs []forgev1alpha1.Registry) ([]config.Registry, error) {
-	var configs []config.Registry
-	for _, apiReg := range apiRegs {
-		regConf := config.Registry{
-			Host:   apiReg.Server,
-			NonSSL: apiReg.NonSSL,
-		}
-
-		// NOTE: move BasicAuth validation into an admission webhook at a later time
-		if err := apiReg.BasicAuth.Validate(); err != nil {
-			return nil, errors.Wrap(err, "basic auth validation failed")
-		}
+// +kubebuilder:rbac:groups=forge.dominodatalab.com,resources=containerimagebuilds,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=forge.dominodatalab.com,resources=containerimagebuilds/status,verbs=get;update;patch
 
-		switch {
-		case apiReg.BasicAuth.IsInline():
-			regConf.Username = apiReg.BasicAuth.Username
-			regConf.Password = apiReg.BasicAuth.Password
-		case apiReg.BasicAuth.IsSecret():
-			var err error
-			regConf.Username, regConf.Password, err = r.getDockerAuthFromSecret(ctx, apiReg.Server, apiReg.BasicAuth.SecretName, apiReg.BasicAuth.SecretNamespace)
-			if err != nil {
-				return nil, err
-			}
-		}
+func (r *ContainerImageBuildReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	ctx := context.Background()
+	log := r.Log.WithValues("containerimagebuild", req.NamespacedName)
 
-		configs = append(configs, regConf)
+	// attempt to load resource by name and ignore not-found errors
+	build := &forgev1alpha1.ContainerImageBuild{}
+	if err := r.Get(ctx, req.NamespacedName, build); err != nil {
+		log.Error(err, "Unable to find resource")
+		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
-	return configs, nil
-}
-
-func (r *ContainerImageBuildReconciler) getDockerAuthFromSecret(ctx context.Context, host, name, namespace string) (string, string, error) {
-	// ctx is currently unused: https://github.com/kubernetes/kubernetes/pull/87299
-	secret, err := r.Clientset.CoreV1().Secrets(namespace).Get(name, metav1.GetOptions{})
+	existing := &batchv1.Job{}
+	err := r.Get(ctx, types.NamespacedName{Name: build.Name, Namespace: build.Namespace}, existing)
 	if err != nil {
-		return "", "", errors.Wrap(err, "cannot find registry auth secret")
-	}
+		// requeue when get job fails
+		if !apierrors.IsNotFound(err) {
+			log.Error(err, "Failed to get build job")
+			return ctrl.Result{}, err
+		}
 
-	if secret.Type != corev1.SecretTypeDockerConfigJson {
-		return "", "", fmt.Errorf("registry auth secret must be %v, not %v", corev1.SecretTypeDockerConfigJson, secret.Type)
-	}
+		if err := r.checkPrerequisites(ctx, build); err != nil {
+			log.Error(err, "Failed to create job prerequisites", "Name", build.Name, "Namespace", build.Namespace)
+			return ctrl.Result{}, err
+		}
```

Comment:
Brings up a question about failed CIB cleanup, but I think we're deferring that for a later time anyway.

Reply:
I think that's correct. This just makes sure we have cascading deletes enabled for when we get there. Just thinking aloud, there may be a way to run a periodic clean job at some predefined cadence or perhaps some other in-reconcile check we can perform to maintain a limit.